rfc/0009-rancher-integration-charts.md

```diff
@@ -0,0 +1,174 @@
+version numbers (that prepend `100-`). All needed changes for Rancher
+Charts must be present in upstream Kubewarden charts.
+If, in the future, we would need functionality from the Rancher Packages
+scripts, we will always have the possibility to add a new stage for it.
+
+The Rancher charts & Rancher packages specification can be read from:
+- [Rancher charts docs](https://rancher.com/docs/rancher/v2.6/en/helm-charts/creating-apps/#additional-files-for-rancher-charts)
+- [Rancher chart dev docs](https://github.com/rancher/charts/blob/dev-v2.6/README.md)
+- [Requirements and chart annotations docs](https://confluence.suse.com/pages/viewpage.action?spaceKey=EN&title=Feature+Charts+Helm+Requirements)
+- [Example of superchart](https://github.com/rancher/charts/tree/dev-v2.6/packages/rancher-monitoring/rancher-monitoring/generated-changes/dependencies)
+
+Modify our charts to meet those. This includes (but is not exhaustive to, as it
+is a moving target):
+
+- Set minimum required Rancher chart annotations:
+  * `catalog.cattle.io/namespace`: should be postfixed with `-system` for UI
+    filtering. When creating v2 of the chart, the namespace should be different
+    than on v1, to handle upgrades/cleanup independently.
+  * `catalog.cattle.io/release-name`: If unset, release name can be changed in UI.
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/certified`
+  * `catalog.cattle.io/experimental`
+  * `catalog.cattle.io/ui-component`
+  * `catalog.cattle.io/display-name`
+  * `catalog.cattle.io/os`
+- Set `catalog.cattle.io/rancher-version` annotation as is required for
+  user-facing charts.
+- Provide `chart/appreadme.md`.
+- Provide logo for the chart by settings `icon` in `Chart.yaml`.
+- Rancher automatically creates a list of container images from charts for
+  airgap. Adopt Rancher's `repository` and `tag` values in values.yml.
+- Provide our own `serviceaccount`.
+- Define k8s min/max version with `kubeVersion` in `Chart.yaml`.
+- Add scheduling options at a pod (affinity) and node level
+  (taints/tolerations/selectors).
+- Provide `annotations.catalog.cattle.io/upstream-version`. This could be
+  automatically bumped on release.
+- Provide `questions.yaml` for defaults on Rancher Explorer UI.
+
+The Kubewarden stack has the following dependencies:
+- Designated main chart: `kubewarden-controller`. This will be the main charts
+  to build a Rancher package from.
+- Hard dependencies: 
+  * CRDs: in `kubewarden-crds` chart. On the main Kubewarden chart
+    (`kubewarden-controller`), set `catalog.cattle.io/auto-install:
+    kubewarden-crds=match` that points to the right CRD version.
+    This annotation accepts `match`, or a hardcoded version (e.g: `1.1.0`)
+    We ought to set it to a hardcoded version, to be able to release patch
+    versions.  of `kubewarden-controller` independently.
+  * `cert-manager`: considered a precondition of Rancher Explorer and out of
+    its scope.
```

Comment:
How is this going to be installed then?

Reply:
Rancher depends on `cert-manager`, and expects the user to have it installed before starting with the installation of Rancher. Hence, we can assume it is present for Kubewarden.

Reply:
Added a note for it in https://github.com/kubewarden/rfc/pull/11/commits/914babd85855ed860df743e6091cdc093e244493.